### PR TITLE
State file version bump and unit test

### PR DESCRIPTION
**Pull request by yunhee-l**
### Summary
State file version bump with envfile task resource support

### Implementation details
- state file version bump
- added envfiles to supported task resource type
- unmarshal method for envfile resource

### Testing

New tests cover the changes: Unit test added 

### Description for the changelog
N/A

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/statemanager/state_manager_test.go [hunk 1]
```diff
@@ -483,6 +483,32 @@ func TestLoadsDataSeqTaskManifest(t *testing.T) {
...
+	)
+	assert.NoError(t, err)
+	err = stateManager.Load()
+	assert.NoError(t, err)
+	tasks, err := taskEngine.ListTasks()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(tasks))
+	assert.Equal(t, "state-file", cluster)
 	assert.EqualValues(t, 0, sequenceNumber)
-	assert.EqualValues(t, 7, seqNumTaskManifest)
```
Comment (sharanyad) [inline agent/statemanager/state_manager_test.go]: I think this is from previous test that got removed?

Comment (yunhee-l) [reply agent/statemanager/state_manager_test.go]: Fixed!

File: agent/statemanager/testdata/v27/environmentFiles/ecs_agent_data.json [hunk 2]
```diff
@@ -0,0 +1,168 @@
+{
+  "Data": {
+  "Cluster": "state-file",
+  "ContainerInstanceArn": "arn:aws:ecs:us-west-2:984736093387:container-instance/ea27e41b-c6e4-45a9-a7a0-484c95abece7",
```
Comment (fenxiong) [inline agent/statemanager/testdata/v27/environmentFiles/ecs_agent_data.json]: maybe change account id/task id/instance id to dummy value

Comment (yunhee-l) [reply agent/statemanager/testdata/v27/environmentFiles/ecs_agent_data.json]: Fixed by replacing with dummy id and garbling the sha etc.

**Pull request merged**